cf-custom-resources/lib/wkld-custom-domain.js

```diff
@@ -0,0 +1,554 @@
+  envName,
+  serviceName,
+  certificateDomain,
+  domainTypes,
+  rootDNSRole,
+  domainName;
+let defaultSleep = function (ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+let sleep = defaultSleep;
+
+const appRoute53Context = () => {
+  let client;
+  return () => {
+    if (!client) {
+      client = new AWS.Route53({
+        credentials: new AWS.ChainableTemporaryCredentials({
+          params: { RoleArn: rootDNSRole },
+          masterCredentials: new AWS.EnvironmentCredentials("AWS"),
+        }),
+      });
+    }
+    return client;
+  };
+};
+
+const envRoute53Context = () => {
+  let client;
+  return () => {
+    if (!client) {
+      client = new AWS.Route53();
+    }
+    return client;
+  };
+};
+
+const acmContext = () => {
+  let client;
+  return () => {
+    if (!client) {
+      client = new AWS.ACM();
+    }
+    return client;
+  };
+};
+
+const resourceGroupsTaggingAPIContext = () => {
+  let client;
+  return () => {
+    if (!client) {
+      client = new AWS.ResourceGroupsTaggingAPI();
+    }
+    return client;
+  };
+};
+
+const clients = {
+  app: {
+    route53: appRoute53Context(),
+  },
+  root: {
+    route53: appRoute53Context(),
+  },
+  env: {
+    route53: envRoute53Context(),
+  },
+  acm: acmContext(),
+  resourceGroupsTaggingAPI: resourceGroupsTaggingAPIContext(),
+};
+
+const appHostedZoneIDContext = () => {
+  let id;
+  return async () => {
+    if (!id) {
+      id = await hostedZoneIDByName(`${appName}.${domainName}`);
+    }
+    return id;
+  };
+};
+
+const rootHostedZoneIDContext = () => {
+  let id;
+  return async () => {
+    if (!id) {
+      id = await hostedZoneIDByName(`${domainName}`);
+    }
+    return id;
+  };
+};
+
+let hostedZoneID = {
+  app: appHostedZoneIDContext(),
+  root: rootHostedZoneIDContext(),
+};
+
+/**
+ * Upload a CloudFormation response object to S3.
+ *
+ * @param {object} event the Lambda event payload received by the handler function
+ * @param {object} context the Lambda context received by the handler function
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+function report(
+  event,
+  context,
+  responseStatus,
+  physicalResourceId,
+  responseData,
+  reason
+) {
+  return new Promise((resolve, reject) => {
+    const https = require("https");
+    const { URL } = require("url");
+
+    let reasonWithLogInfo = `${reason} (Log: ${context.logGroupName}/${context.logStreamName})`;
+    let responseBody = JSON.stringify({
+      Status: responseStatus,
+      Reason: reasonWithLogInfo,
+      PhysicalResourceId: physicalResourceId || context.logStreamName,
```

Comment:
(ik this is not your doing but) do you think it's safe that we just remove the `|| context.logStreamName` part since it seems like `physicalResourceId` will never be empty?